src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
```

Comment:
The function `arrayDateFromObject` is also not defined in the provided code. This function is crucial for creating a date object from the array and format provided.